storagemarket/impl/provider_states.go

```diff
@@ -107,7 +107,7 @@ func (p *Provider) verifydata(ctx context.Context, deal MinerDeal) (func(*MinerD
 	allSelector := ssb.ExploreRecursive(selector.RecursionLimitNone(),
 		ssb.ExploreAll(ssb.ExploreRecursiveEdge())).Node()
 
-	commp, file, err := p.pio.GeneratePieceCommitment(deal.Ref, allSelector)
+	commp, file, _, err := p.pio.GeneratePieceCommitment(deal.Ref, allSelector)
```

Comment:
So this here is the main issue with this PR -- the file is read by someone else who expects it to have the length of the whole piece.

Reply:
Should we be storing .car files that are padded then?

In the offline dealflow, what we thought we should do is create .car files (no padding), calculate and store CommP, and then send those files to miners. Then the miners would need to calculate and generate CommP again to match. Would it be more efficient for us to just do the padding when we create the .car files, pass a padded file into a function like this, and then send the padded files to the miners?

Reply:
we resolved this else and this PR is merged :)